storage/src/mmr/verification.rs

```diff
@@ -425,57 +465,77 @@ impl<D: Digest> Proof<D> {
     }
 }
 
-fn peak_digest_from_range<'a, I, H, E, S>(
-    hasher: &mut H,
+/// Information about the current range of nodes being traversed.
+struct RangeInfo {
     pos: u64,           // current node position in the tree
     two_h: u64,         // 2^height of the current node
     leftmost_pos: u64,  // leftmost leaf in the tree to be traversed
     rightmost_pos: u64, // rightmost leaf in the tree to be traversed
+}
+
+fn peak_digest_from_range<'a, I, H, E, S>(
+    hasher: &mut H,
+    range_info: RangeInfo,
     elements: &mut E,
     sibling_digests: &mut S,
+    mut collected_digests: Option<&mut Vec<(u64, I::Digest)>>,
```

Comment:
I haven't tried this but does it make more sense to do: &mut Option<Vec<(u64, I::Digest)>>?